### PR TITLE
Review Dependabot alerts for `net6.0` and `net7.0` target frameworks and bump all packages

### PR DESCRIPTION
**Pull request by raman-m**
## Fixes
- #2218 

## Proposed Changes
  - Updated all packages to their latest versions
  - Removed the `net7.0` target framework from testing projects due to NETSDK1138 warning (indicating that .NET 7 is out of support)
  - Resolved warnings and messages

### P.S.
It was impossible to address [Dependabot alerts](https://github.com/ThreeMammals/Ocelot/security/dependabot) for .NET6-7 because of IdentityServer4 requires significant development efforts; therefore, we eliminate IS4 while upgrading to NET 9. Additionally, migrating from IdentityServer4 to [Duende.IdentityServer](https://www.nuget.org/packages/Duende.IdentityServer) is inappropriate due to its non-OSS status (and I don't want to advertise it). It is preferable to migrate testing projects to the [ASP.NET Core Identity](https://learn.microsoft.com/en-us/aspnet/core/security/authentication/identity?view=aspnetcore-9.0&tabs=visual-studio) framework.

### CODE REVIEW

File: src/Ocelot.Cache.CacheManager/Ocelot.Cache.CacheManager.csproj [hunk 1]
```diff
@@ -43,18 +43,18 @@
...
+    <PackageReference Include="Microsoft.Extensions.Logging" Version="6.0.1" />
     <PackageReference Include="Microsoft.NETCore.Platforms" Version="6.0.13" />
   </ItemGroup>
   <!-- Conditionally obtain references for the net 7.0 target -->
   <ItemGroup Condition=" '$(TargetFramework)' == 'net7.0' ">
-    <PackageReference Include="Microsoft.Extensions.Configuration" Version="7.0.0" />
-    <PackageReference Include="Microsoft.Extensions.Configuration.Binder" Version="7.0.4" />
-    <PackageReference Include="Microsoft.Extensions.DependencyInjection" Version="7.0.0" />
-    <PackageReference Include="Microsoft.Extensions.Logging" Version="7.0.0" />
+    <PackageReference Include="Microsoft.Extensions.Configuration" Version="8.0.0" />
```
Comment (ggnaegi) [inline src/Ocelot.Cache.CacheManager/Ocelot.Cache.CacheManager.csproj:54]: Are you sure about this? You're targeting net7.0...

File: src/Ocelot.Cache.CacheManager/Ocelot.Cache.CacheManager.csproj [hunk 2]
```diff
@@ -43,18 +43,18 @@
...
+    <PackageReference Include="Microsoft.Extensions.Configuration.Binder" Version="8.0.2" />
+    <PackageReference Include="Microsoft.Extensions.DependencyInjection" Version="8.0.1" />
+    <PackageReference Include="Microsoft.Extensions.Logging" Version="8.0.1" />
     <PackageReference Include="Microsoft.NETCore.Platforms" Version="7.0.4" />
   </ItemGroup>
   <!-- Conditionally obtain references for the net 8.0 target -->
```
Comment (ggnaegi) [inline src/Ocelot.Cache.CacheManager/Ocelot.Cache.CacheManager.csproj:60]: You might want to update the .NET 8 dependencies too...

File: src/Ocelot.Cache.CacheManager/Ocelot.Cache.CacheManager.csproj (hunk 1 above)
Comment (raman-m) [reply src/Ocelot.Cache.CacheManager/Ocelot.Cache.CacheManager.csproj:54]: Yes, I am. Could you check this link: [Microsoft.Extensions.Configuration/8.0.0 | Frameworks](https://www.nuget.org/packages/Microsoft.Extensions.Configuration/8.0.0#supportedframeworks-body-tab)
image
So, `net7.0` is included into the list.

File: src/Ocelot.Cache.CacheManager/Ocelot.Cache.CacheManager.csproj (hunk 2 above)
Comment (raman-m) [reply src/Ocelot.Cache.CacheManager/Ocelot.Cache.CacheManager.csproj:60]: Now we have consistency with testing projects where .NET8 refs are updated to 9.0.0
Fixed in commit https://github.com/ThreeMammals/Ocelot/pull/2220/commits/fed86f70cde854902900ff430482491d89937de0 ✔️

File: src/Ocelot.Cache.CacheManager/Ocelot.Cache.CacheManager.csproj (hunk 1 above)
Comment (raman-m) [reply src/Ocelot.Cache.CacheManager/Ocelot.Cache.CacheManager.csproj:54]: The strategy is straightforward: search for the most updated packs by date that have the max version number and supporting the current target framework. Therefore:
- All .NET 7 references have updated to version `8.0.x`, targets `net7.0`
- All .NET 8 references have been updated to version `9.0.0`, which targets `net8.0`

Is it now clear? For further information please review the updated test projects where the same strategy has applied.

**Review by ggnaegi — APPROVED** [review]

**Pull request merged**